### ISSUE TITLE
"Windows fatal exception: access violation" when calling `decimate`

### ISSUE THREAD

**Issue by nidu** [pyvista/pyvista-support#229]
## Description

When calling a series of delaunay, clip_box and decimate on a point cloud - i'm getting "Windows fatal exception: access violation" sometimes, more precisely:

```
Windows fatal exception: access violation

Current thread 0x00003ae8 (most recent call first):
  File "C:\Users\kally\anaconda3\envs\geo\lib\site-packages\pyvista\core\filters.py", line 47 in _update_alg
  File "C:\Users\kally\anaconda3\envs\geo\lib\site-packages\pyvista\core\filters.py", line 2914 in decimate
```

--------------------------------------------------------------------------------
  Date: Tue Aug 11 23:14:16 2020 W. Europe Daylight Time

                OS : Windows
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
       Environment : Python

  Python 3.7.7 (default, May  6 2020, 11:45:54) [MSC v.1916 64 bit (AMD64)]

             numpy : 1.18.5
             scipy : 1.5.0
           IPython : 7.16.1
        matplotlib : 3.2.2
            scooby : 0.5.6

  Intel(R) Math Kernel Library Version 2020.0.1 Product Build 20200208 for
  Intel(R) 64 architecture applications
--------------------------------------------------------------------------------

## Example Data

I can reproduce it with this file:

```python
import pyvista as pv

print('Generate hills')
m = pv.ParametricRandomHills().extract_geometry()
print(f'Triangulate {m.n_points} points')
m = m.delaunay_2d()
print(f'Clip box {m.n_points} points, {m.n_faces} faces')
m = m.clip_box((-10, 0, -10, 0, -10, 0), invert=False).extract_geometry()
print(f'Decimate {m.n_points} points, {m.n_faces} faces')
m = m.decimate(.99)
```

The output is

```
Generate hills
Triangulate 10000 points
Clip box 10000 points, 19643 faces
Decimate 156 points, 151 faces
Windows fatal exception: access violation

Current thread 0x00000d74 (most recent call first):
  File "C:\Users\kally\anaconda3\envs\geo\lib\site-packages\pyvista\core\filters.py", line 47 in _update_alg
  File "C:\Users\kally\anaconda3\envs\geo\lib\site-packages\pyvista\core\filters.py", line 2914 in decimate
  File ".\error_report.py", line 10 in <module>
```

**Comment by banesullivan** [comment]
This seems related to https://github.com/pyvista/pyvista/issues/861... @pyvista/developers, this is concerning that we've had two Windows users recently report the `Windows fatal exception: access violation` error.

@nidu, can you please share a PyVsita report (you shared a `scooby` default report which doesn't show details about PyVista)? `print(pv.Report(gpu=False))` (`gpu=False` to avoid any errors)

**Comment by banesullivan** [comment]
My immediate advice, I suspect you are using the latest version of VTK (9.x). Please try downgrading VTK to 8.1.2 and see if the error happens still

**Comment by nidu** [comment]
Here's the info from `pv.Report(gpu=False)`

--------------------------------------------------------------------------------
  Date: Fri Aug 14 07:59:24 2020 W. Europe Daylight Time

                OS : Windows
            CPU(s) : 12
           Machine : AMD64
      Architecture : 64bit
       Environment : Python
       GPU Details : None

  Python 3.7.7 (default, May  6 2020, 11:45:54) [MSC v.1916 64 bit (AMD64)]

           pyvista : 0.25.3
               vtk : 8.2.0
             numpy : 1.18.5
           imageio : 2.9.0
           appdirs : 1.4.3
            scooby : 0.5.6
            meshio : 4.0.16
        matplotlib : 3.2.2
             PyQt5 : 5.12.3
           IPython : 7.16.1
             scipy : 1.5.0

  Intel(R) Math Kernel Library Version 2020.0.1 Product Build 20200208 for
  Intel(R) 64 architecture applications
--------------------------------------------------------------------------------

**Comment by RichardScottOZ** [comment]
I've got one

Windows fatal exception: access violation

Current thread 0x00006208 (most recent call first):
  File "C:\Users\rscott\AppData\Local\Continuum\anaconda3\envs\gempy\lib\site-packages\pyvista\plotting\renderer.py", line 1090 in reset_camera
  File "C:\Users\rscott\AppData\Local\Continuum\anaconda3\envs\gempy\lib\site-packages\pyvista\plotting\renderer.py", line 377 in add_actor
  File "C:\Users\rscott\AppData\Local\Continuum\anaconda3\envs\gempy\lib\site-packages\pyvista\plotting\plotting.py", line 335 in add_actor
  File "C:\Users\rscott\AppData\Local\Continuum\anaconda3\envs\gempy\lib\site-packages\pyvista\plotting\plotting.py", line 1354 in add_mesh
  File "C:\Users\rscott\AppData\Local\Continuum\anaconda3\envs\gempy\lib\site-packages\pyvista\plotting\helpers.py", line 113 in plot
  File "test.py", line 187 in <module>

I think it is related to faces though, e.g. from the tsurf test - I haven't checked them to make sure they all make sense, but not your usual error

**Comment by RichardScottOZ** [comment]
e.g. https://github.com/pyvista/pyvista/issues/4902

**Comment by RichardScottOZ** [comment]
yeah, this is apparently because need another dimension to say the triangle is a triangle at the start!
